tracer/test/Datadog.Trace.Tests/Util/Http/QueryStringObfuscatorTests.cs

```diff
@@ -0,0 +1,77 @@
+            result.Should().Be(originalQueryString);
+        }
+
+        [Theory]
+        [InlineData(null)]
+        [InlineData("")]
+        [InlineData(" ")]
+        public void EdgeCases(string querystring)
+        {
+            var logger = new Mock<IDatadogLogger>();
+            var queryStringObfuscator = new Obfuscator(TracerSettings.DefaultObfuscationQueryStringRegex, _timeout, logger.Object);
+            var result = queryStringObfuscator.Obfuscate(querystring);
+            result.Should().Be(querystring);
+        }
+
+        [Fact]
+        public void ObfuscateWithDefaultPattern()
+        {
+            var logger = new Mock<IDatadogLogger>();
+            var queryStringObfuscator = new Obfuscator(TracerSettings.DefaultObfuscationQueryStringRegex, _timeout, logger.Object);
+            var queryString = "key1=val1&token=a0b21ce2-006f-4cc6-95d5-d7b550698482&key2=val2";
+            var result = queryStringObfuscator.Obfuscate(queryString);
+            result.Should().Be("key1=val1&<redacted>&key2=val2");
+        }
+
+        [Fact]
+        public void ObfuscateWithCustomPattern()
+        {
+            var logger = new Mock<IDatadogLogger>();
+            var queryStringObfuscator = new Obfuscator(@"(?i)(?:p(?:ass)?w(?:or)?d|pass(?:_?phrase)?|secret|(?:api_?|private_?|public_?|access_?|secret_?)key(?:_?id)?|authentic\d*)(?:\s*=[^&]+|""\s*:\s*""[^""]+"")|[a-z0-9\._\-]{100,}", _timeout, logger.Object);
+            var queryString = "?authentic1=val1&token=a0b21ce2-006f-4cc6-95d5-d7b550698482&key2=val2&authentic3=val2&authentic55=v";
+            var result = queryStringObfuscator.Obfuscate(queryString);
+            result.Should().Be("?<redacted>&token=a0b21ce2-006f-4cc6-95d5-d7b550698482&key2=val2&<redacted>&<redacted>");
+        }
+
+        [Fact]
+        public void ObfuscateTimeout()
+        {
+            var logger = new Mock<IDatadogLogger>();
+            var querystringObfuscator = new Mock<Obfuscator>(() => new Obfuscator(@"\[(.*?)\]((?:.\s*)*?)\[\/\1\]", _timeout, logger.Object));
+            logger.Setup(q => q.Error(It.IsAny<RegexMatchTimeoutException>(), It.IsAny<string>(), It.IsAny<double>(), It.IsAny<string>(), It.IsAny<int>(), It.IsAny<string>())).Verifiable();
+            const string queryString = @"?[tag1]Test's Text Test Text Test Text Test Text.Test Text Test Text Test ""Text Test Text"" Test Text Test Text.Test Text ? Test Text Test Text Test Text Test Text Test Text Test Text.Test Text, Test Text Test Text Test Text Test Text Test Text Test Text Test Text.[/ ta.g1][tag2]Test's Text Test Text Test Text Test Text.Test Text Test Text Test ""Text Test Text"" Test Text Test Text.Test Text ? Test Text Test Text Test Text Test Text Test Text Test Text.Test Text, Test Text Test Text Test Text Test Text Test Text Test Text Test Text.[/ tag2]";
+            var result = querystringObfuscator.Object.Obfuscate(queryString);
```

Comment:
I'm a little concerned that this test could be flaky if for some reason the hardware is really good. Do you think that's a concern? Should we set the timeout to a super low value as well?